exporter.go

```diff
@@ -62,11 +63,44 @@ func getEnvDuration(key string, defaultValue time.Duration) time.Duration {
 	return d
 }
 
-func createClientWithRetries(getClient func() (interface{}, error), retries int, retryInterval time.Duration) (interface{}, error) {
+// positiveDuration is a wrapper of time.Duration to ensure only positive values are accepted
+type positiveDuration struct{ duration *time.Duration }
+
+func (pd positiveDuration) String() string {
+	return fmt.Sprint(pd.duration)
+}
+
+func (pd positiveDuration) Set(s string) error {
+	if pd.isNegative(s) {
+		return fmt.Errorf("Negative duration is not valid")
+	}
+	dur, err := time.ParseDuration(s)
+	if err != nil {
+		return err
+	}
+
+	*pd.duration = dur
+	return nil
+}
+
+func (pd positiveDuration) isNegative(s string) bool {
+	return s[0] == '-'
+}
+
+func createPositiveDurationFlag(dur time.Duration, key, helper string) error {
+	pd := positiveDuration{&dur}
+	if pd.isNegative(dur.String()) {
```

Comment:
this is not required. let's assume that the value passed to `createPositiveDurationFlag` is correct, as the other `flag.*` functions assume. In our case, in case we won't have an invalid default value,  because  the suggested `getEnvPositiveDuration` will log.Fatalf in that case.